apps/dcos_l4lb/src/dcos_l4lb_ipset_mgr.erl

```diff
@@ -0,0 +1,353 @@
+-module(dcos_l4lb_ipset_mgr).
+
+-include_lib("gen_netlink/include/netlink.hrl").
+
+-behaviour(gen_server).
+
+%% API
+-export([
+    start_link/0,
+    get_entries/0,
+    get_entries/1,
+    add_entries/2,
+    remove_entries/2,
+    cleanup/0
+]).
+
+%% gen_server callbacks
+-export([init/1, handle_continue/2,
+    handle_call/3, handle_cast/2]).
+
+-export_type([entry/0]).
+
+-type entry() :: dcos_l4lb_mesos_poller:key().
+
+-record(state, {
+    netlink :: pid()
+}).
+
+-define(IPSET_PROTOCOL, 6).
+-define(IPSET_TYPE, "hash:ip,port").
+-define(IPSET_NAME, "dcos-l4lb").
+-define(IPSET_NAME_IPV6, "dcos-l4lb-ipv6").
+
+
+-spec(get_entries() -> [entry()]).
+get_entries() ->
+    % NOTE: FOR DEBUG PURPOSE ONLY.
+    {ok, Pid} = start_link(),
+    try
+        get_entries(Pid)
+    after
+        erlang:unlink(Pid),
+        gen_server:cast(Pid, stop)
+    end.
+
+-spec(get_entries(pid()) -> [entry()]).
+get_entries(Pid) ->
+    gen_server:call(Pid, get_entries).
+
+-spec(add_entries(pid(), [entry()]) -> ok).
+add_entries(Pid, Entries) ->
+    gen_server:call(Pid, {add_entries, Entries}).
+
+-spec(remove_entries(pid(), [entry()]) -> ok).
+remove_entries(Pid, Entries) ->
+    gen_server:call(Pid, {remove_entries, Entries}).
+
+-spec(start_link() ->
+    {ok, Pid :: pid()} | ignore | {error, Reason :: term()}).
+start_link() ->
+    gen_server:start_link(?MODULE, [], []).
+
+%%%===================================================================
+%%% gen_server callbacks
+%%%===================================================================
+
+init([]) ->
+    {ok, {}, {continue, dcos_l4lb_config:ipset_enabled()}}.
+
+handle_continue(false, {}) ->
+    {noreply, disabled};
+handle_continue(true, {}) ->
+    {ok, Pid} = gen_netlink_client:start_link(?NETLINK_NETFILTER),
+    {ok, ?IPSET_PROTOCOL} = get_protocol_version(Pid),
+    {ok, []} = maybe_create_ipset(Pid, inet),
+    {ok, _} = maybe_add_iptables_rules(inet),
+    {ok, []} = maybe_create_ipset(Pid, inet6),
+    {ok, _} = maybe_add_iptables_rules(inet6),
+    {noreply, #state{netlink=Pid}}.
+
+handle_call(get_entries, _From, disabled) ->
+    {reply, [], disable};
```

Comment:
JFI I've been testing disabled ipset and found a typo.

Reply:
Thanks @urbanserj. Should have been caught in review 🤦‍♂️

Reply:
No, it's my bad.